src/Evolution/Executables/Cce/KleinGordonCharacteristicExtract.hpp

```diff
@@ -0,0 +1,105 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Evolution/Executables/Cce/CharacteristicExtractBase.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/BouncingBlackHole.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/GaugeWave.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/LinearizedBondiSachs.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/RobinsonTrautman.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/RotatingSchwarzschild.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp"
+#include "Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.hpp"
+#include "Evolution/Systems/Cce/Components/KleinGordonCharacteristicEvolution.hpp"
+#include "Evolution/Systems/Cce/Events/ObserveFields.hpp"
+#include "Evolution/Systems/Cce/Events/ObserveTimeStep.hpp"
+#include "Evolution/Systems/Cce/Initialize/ConformalFactor.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "Parallel/Algorithms/AlgorithmSingleton.hpp"
+#include "ParallelAlgorithms/Events/Factory.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/EventsAndTriggers.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/LogicalTriggers.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Trigger.hpp"
+#include "Time/StepChoosers/Factory.hpp"
+#include "Time/TimeSteppers/Factory.hpp"
+#include "Time/Triggers/TimeTriggers.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+template <template <typename> class BoundaryComponent>
+struct EvolutionMetavars : CharacteristicExtractDefaults<false> {
```

Comment:
Have you tried compiling this executable yet?

Reply:
Yes, it works. But thinking more carefully about it, I feel it's more convenient to include all the actions of `CharacteristicEvolution<Metavariables>` in this commit, since the GR (tensor) part of the two executables should be the same. Actions for the scalar part will be added later. I've just pushed one more commit that does this. For the moment, the executable `KleinGordonCharacteristicEvolution` does the same thing as `CharacteristicEvolution`, except that `Actions::RequestBoundaryData`, `Actions::ReceiveWorldtubeData`, `Actions::RequestNextBoundaryData` are not added, because they are not ready for the `KleinGordonH5WorldtubeBoundary` boundary component. If you are happy with this change, I will squash it with the second commit.